### ISSUE TITLE
Dispatcher does not emit "finish" event.

### ISSUE THREAD

**Issue by FireController1847** [discordjs/discord.js#2620]
**Please describe the problem you are having in as much detail as possible:**
The StreamDispatcher is not emitting the stream event.

**Include a reproducible code sample here, if possible:**
```js
// Evalling this snippet on my bot did not send the "end" or "finish" messages.
m.member.voiceChannel.join().then(con => {
  const stream = yt("https://www.youtube.com/watch?v=e0ykswI49BI", { quality: "highestaudio" } );
  const dis = con.play(stream);
  dis.on("end", () => {
    m.channel.send("end");
  });
  dis.on("finish", () => {
    m.channel.send("finish");
  });
  global.dis = dis;
  global.con = con;
  global.ytstream = stream;
}).catch(console.error).then(() => {
  m.channel.send("hi");
});
```
```js
// Adding an event to the stream, the only message sent after the bot finished was "stream ended". Obviously something is going on on Discord.js's end.
m.member.voiceChannel.join().then(con => {
  const stream = yt("https://www.youtube.com/watch?v=e0ykswI49BI", { quality: "highestaudio" } );
  const dis = con.play(stream);
  dis.on("end", () => {
    m.channel.send("end");
  });
  dis.on("finish", () => {
    m.channel.send("finish");
  });
  stream.on("finish", () => {
    m.channel.send("stream ended");
  });
  global.dis = dis;
  global.con = con;
  global.ytstream = stream;
}).catch(console.error).then(() => {
  m.channel.send("hi");
});
```

**Further details:**

- discord.js version: v12 3d41748a3802aed678bddccf76c9e5d2baa48c1d
- node.js version: v10.1.0
- Operating system: Ubuntu 18.04
- Priority this issue should have – please be realistic and elaborate if possible: Pretty high, this is quite important for almost all queue and voice related features.

- [ ] I found this issue while running code on a __user account__
- [x] I have also tested the issue on latest master, commit hash: 3d41748a3802aed678bddccf76c9e5d2baa48c1d

**Comment by hydrabolt** [comment]
Can you try installing the development version of prism, `npm i hydrabolt/prism-media` to see if that fixes it?

**Comment by FireController1847** [comment]
Yes, that did fix it!

**Comment by hydrabolt** [comment]
I've just released prism 0.3.1 which fixes this for all users!
